docs/documentation/automation/admin_cotlang.md

```diff
@@ -7,18 +7,236 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
 
-# WIP
+# Overview {#cotlang-overview}
 
-# Cotlang Overview {#cotlang-overview}
+Script language created by Cotalker.
 
-Language created by Cotalker. <br/>
-
-It is used to extract data from different contexts in Cotalker. It is useful to configure the routine.
+It is used to extract data from different contexts in Cotalker. It is useful for configuring routines.
 
 ## Data Context {#data-context}
-This table contains the data to which the different sources have access. For example, if you are using cotlang in a survey trigger routine, you will have direct access to the task. <br/>
+This table contains the data to which the different sources have access. For example, if you are using Cotlang in a survey trigger routine, you will have direct access to the task.
+
+<div className="row table-row-1">
+
+### Global Message Trigger {#global-message-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    channel: COTChannel,
+    message: COTMessage,
+    cmdArgs: Array[string]
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: _The source is a global bot that is triggered with a slash command:_ `/command`
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Channel Message Trigger {#channel-message-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+  channel: COTChannel,
+  message: COTMessage,
+  cmdArgs: Array[string]
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: _The source is a bot that is triggered with a slash command and is associated with particular channels:_ `/command`
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Global Survey Trigger {#global-survey-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+  ...COTAnswer,
+  messages: COTMessage
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
 
-#### Coming soon!
+**Note**: _The source is a bot that is triggered with a slash command and is associated with particular channels:_ `/command`
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Channel Survey Trigger {#channel-survey-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    ...COTAnswer,
+    messages: COTMessage
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: _Messages have to be active to use them. The source is a bot that is triggered with a survey and is associated with particular channels._
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM Survey Trigger {#sm-survey-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    ...COTTask,
+    sentAnswer: COTAnswer
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM Change State {#sm-change-state}
```

Comment:
What do you thing about replacing `SM` to `State machine`

Reply:
Good question. Actually, I was thinking to replace `SM`with `Workflow`, since we had agreed to use "workflow" instead of "state machine" a while back.

Reply:
They are generally interchangeable, but beware we still have to concepts (we can re-brand them)
On the other side, we also require to match the admin names and tags with the documentation.
If we change a name, we need to coordinate with engineering to change the names in the admin as well (you can also send your own PRs if you want to change something in the webclient-admin)

E.g., This Workflow - has 8 state machines.

![image](https://user-images.githubusercontent.com/3951895/115402315-50b51b00-a1b9-11eb-9591-e28db2cc415b.png)

Reply:
I beg to differ, but according to the display and documentation, this a *workflow group* with eight *workflows*.

Reply:
That is right! 
it doesn't say state-machine anywhere.